File: Programmierkurs 1/beispielprogramme/BubbleSort.java
package beispielprogramme;

import java.util.Scanner;

public final class BubbleSort {
    private BubbleSort() {}

    private static final Scanner EINGABE = new Scanner(System.in);

    public static void main(String[] args) {
        int n;
        System.out.print("Anzahl zu sortierender Werte eingeben: ");
        n = EINGABE.nextInt();

        int[] a = new int[n];
        System.out.printf("%d ganze Zahlen eingeben: ", n);

        for (int i = 0; i < a.length; i++) {
            a[i] = EINGABE.nextInt();
        }

        for (int i = a.length; i > 1; i--) {
            for (int j = 0; j < i - 1 ; j++) {
                if (a[j] > a[j + 1]) {
                    int tmp = a[j + 1];
                    a[j + 1] = a[j];
                    a[j] = tmp;
                }
            }
        }

        System.out.print("Sortierte Zahlenfolge: ");

        for (int i = 0; i < a.length ; i++) {
            System.out.printf("%d ", a[i]);
        }

        System.out.println();
    }
}

File: Systemprogrammierung/aufgabe2/bubblesort.c
#include <stdio.h>
#include <stdlib.h>
#include <stddef.h>
#include <time.h>

int main(int argc, char* argv[])
{
    srand((unsigned int)time(NULL));

    int arraysize;
    printf("Anzahl zu sortierender Werte eingeben: ");
    scanf("%d", &arraysize);
    int* a = malloc(arraysize * sizeof(int));
    if (a == NULL)
    {
        printf("Speicherresservierung fehlgeschlagen!\n");
        return 1;
    }
    printf("%d ganze Zahlen eingeben: ", arraysize);

    for (int i = 0; i < arraysize; i++)
    {
        int check = scanf("%d", &a[i]);
        if (check == EOF || check < 0)
        {
            a[i] = rand();
        }
    }
    for (int i = arraysize; i > 1; i--)
    {
        for (int j = 0; j < i - 1; ++j)
        {
            if (a[j] > a[j + 1])
            {
                int tmp = a[j + 1];
                a[j + 1] = a[j];
                a[j] = tmp;
            }
        }
    }
    for (int i = 0; i < arraysize; ++i)
    {
        printf("%d\n", a[i]);
    }
    return 0;
}
